### DISCUSSION TITLE
Unable to obtain fan speed in Python using LibreHardareMonitorLib.dll

### DISCUSSION THREAD

**Discussion by SZPT18010137** [LibreHardwareMonitor/LibreHardwareMonitor#1464] [General]
I am using LibreHardareMonitorLib.dll to get hardware information in Python, but when I list all the hardware sensors value, the fan speed get the None value. Please tell me how to get the fan speed in correctly, and the following is my code:

`
import clr
import os

class HardwareMonitor:

    def __init__(self):
        
        ''' Initial class '''
        
        self.environment_init()
        self.hardware_monitor_init()

    def environment_init(self):
        
        ''' Configure LibreHardwareMonitorLib to environment  '''

        script_dir = os.path.dirname(os.path.abspath(__file__))
        dll_path = os.path.join(script_dir, 'LibreHardwareMonitorLib.dll')
        clr.AddReference(dll_path)

    def hardware_monitor_init(self):

        ''' Configure Hardware monitor '''
        
        from LibreHardwareMonitor import Hardware
        
        self.computer_monitor = Hardware.Computer()

        self.computer_monitor.IsCpuEnabled = True
        self.computer_monitor.IsGpuEnabled = True
        self.computer_monitor.IsBatteryEnabled = True
        #self.computer_monitor.IsControllerEnabled = True
        self.computer_monitor.IsMotherboardEnabled = True
        #self.computer_monitor.IsPsuEnabled = True
        self.computer_monitor.IsStorageEnabled = True
        self.computer_monitor.Open()
        
    def list_message(self):
        index = 0
        for hardware in self.computer_monitor.Hardware:
            hardware.Update()
            
        for hardware in self.computer_monitor.Hardware:
            print(f"Hardware: {hardware.Name}")
            for subhardware  in hardware.SubHardware:
                print(f"\tSubhardware: {subhardware.Name}")
                for sensor in subhardware.Sensors:
                    print(f"\t\tSensor: {sensor.Name}, value: {sensor.Value}")
            for sensor in hardware.Sensors:
                print(f"\tSensor: {sensor.Name}, value: {sensor.Value}")
            
    def test(self):
        
        self.list_message()
        print('Test end')
        
if __name__ == '__main__':
    demo = HardwareMonitor()
    demo.test()
`

And the result of the code print in console:

Hardware: ASUS PRIME H310M-F R2.0
	Subhardware: Nuvoton NCT6798D
		Sensor: Fan #1, value: None
		Sensor: Fan #2, value: None
		Sensor: Fan #3, value: None
		Sensor: Fan #4, value: None
		Sensor: Fan #5, value: None
		Sensor: Fan #6, value: None
		Sensor: Fan #7, value: None
Hardware: Intel Core i3-9100F
	Sensor: CPU Core #1, value: 45.35160827636719
	Sensor: CPU Core #2, value: 47.16666793823242
	Sensor: CPU Core #3, value: 35.54136657714844
	Sensor: CPU Core #4, value: 49.213539123535156
	Sensor: CPU Total, value: 44.318294525146484
	Sensor: CPU Core Max, value: 49.213539123535156
	Sensor: CPU Core #1, value: 59.0
	Sensor: CPU Core #2, value: 56.0
	Sensor: CPU Core #3, value: 60.0
	Sensor: CPU Core #4, value: 59.0
	Sensor: CPU Package, value: 60.0
	Sensor: CPU Core #1 Distance to TjMax, value: 41.0
	Sensor: CPU Core #2 Distance to TjMax, value: 44.0
	Sensor: CPU Core #3 Distance to TjMax, value: 40.0
	Sensor: CPU Core #4 Distance to TjMax, value: 41.0
	Sensor: Core Max, value: 60.0
	Sensor: Core Average, value: 58.5
	Sensor: CPU Core #1, value: 4000.276123046875
	Sensor: CPU Core #2, value: 4000.276123046875
	Sensor: CPU Core #3, value: 4000.276123046875
	Sensor: CPU Core #4, value: 4000.276123046875
	Sensor: CPU Package, value: 27.355520248413086
	Sensor: CPU Cores, value: 23.316574096679688
	Sensor: CPU Memory, value: 0.7935488224029541
	Sensor: CPU Core, value: 1.3031005859375
	Sensor: CPU Core #1, value: 1.205322265625
	Sensor: CPU Core #2, value: 1.278564453125
	Sensor: CPU Core #3, value: 1.305419921875
	Sensor: CPU Core #4, value: 1.305419921875
	Sensor: Bus Speed, value: 100.00690460205078
Hardware: NVIDIA GeForce GT 710
	Sensor: GPU Core, value: 40.0
	Sensor: GPU Core, value: 953.708984375
	Sensor: GPU Memory, value: 499.9989929199219
	Sensor: GPU Fan, value: 47.0
	Sensor: GPU Core, value: 1.0
	Sensor: GPU Memory Controller, value: 8.0
	Sensor: GPU Video Engine, value: 0.0
	Sensor: GPU Bus, value: 0.0
	Sensor: GPU Memory Total, value: 2048.0
	Sensor: GPU Memory Free, value: 196.0
	Sensor: GPU Memory Used, value: 1851.0
	Sensor: GPU Memory, value: 90.38753509521484
Hardware: Lenovo SSD SL700 480G
	Sensor: Temperature, value: 40.0
	Sensor: Used Space, value: 65.97915649414062
	Sensor: Read Activity, value: 2.6101050025317818e-05
	Sensor: Write Activity, value: 1.0188988198933657e-05
	Sensor: Total Activity, value: 99.9998779296875
	Sensor: Read Rate, value: None
	Sensor: Write Rate, value: None
Test end

**Comment by MarkVabulas** [comment]
You need to elevate, it works for me with [elevate](https://pypi.org/project/elevate/)

**Reply by Xcalibur37** [reply]
Also, make sure your fans are direct to the mainboard vs being attached to a 3rd party controller. Ex. I get all 0's because my fans are all routed through Corsair Commander Core XT's. You would need the specific libraries for those (and as far as I can tell, they don't exist as per Corsair).
